src/include/migraphx/output_iterator.hpp

```diff
@@ -64,6 +64,13 @@ function_output_iterator<F> make_function_output_iterator(F f)
     return {std::move(f)};
 }
 
+template <class Container>
+auto join_back_inserter(Container& c)
+{
+    return make_function_output_iterator(
+        [&](const auto& r) { c.insert(c.end(), r.begin(), r.end()); });
+}
```

Comment:
Because its joining the inputs which could be a container of containers of elements to just a container of elements:

```cpp
std::vector<std::vector<int>> input;
std::vector<int> output;
std::copy(input.begin(), input.end(), join_back_inserter(output));
```